src/storage/DataAccessorBasedStore.ts

```diff
@@ -347,6 +398,46 @@ export class DataAccessorBasedStore implements ResourceStore {
     }
   }
 
+  /**
+   * Write the given metadata resource to the DataAccessor.
+   * @param identifier - Identifier of the resource.
+   * @param representation - Corresponding Representation.
+   *
+   * @returns Identifiers of resources that were possibly modified.
+   */
+  protected async writeMetadata(identifier: ResourceIdentifier, representation: Representation):
+  Promise<ResourceIdentifier[]> {
+    const subjectIdentifier = this.metadataStrategy.getSubjectIdentifier(identifier);
+
+    // Cannot create metadata without a corresponding resource
+    if (!await this.hasResource(subjectIdentifier)) {
+      throw new ConflictHttpError('Metadata resources can not be created directly.');
+    }
+
+    // https://github.com/CommunitySolidServer/CommunitySolidServer/issues/1027#issuecomment-988664970
+    // It must not be possible to create .meta.meta resources
+    if (this.metadataStrategy.isAuxiliaryIdentifier(subjectIdentifier)) {
+      throw new ConflictHttpError(
+        'Not allowed to create metadata resources on a metadata resource.',
+      );
+    }
+
+    const metadata = new RepresentationMetadata(subjectIdentifier);
+    let rdf = representation;
+    if (representation.metadata.contentType && representation.metadata.contentType !== INTERNAL_QUADS) {
+      rdf = await this.converter.handleSafe({
+        identifier,
+        representation,
+        preferences: { type: { [INTERNAL_QUADS]: 1 }},
+      });
+    }
+
+    const quads = await arrayifyStream(rdf.data);
+    metadata.addQuads(quads);
```

Comment:
I currently have list of issues that have to be added after this PR. However, it might not be complete so when this PR is merged I will go over all the comments again and check our private conversation to see if there is still missing something.
I think that should be sufficient :smile: